news/13635-add-no-python-user-site-packages

```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Adds a new setting, `isolate-python-env`, that can be used to disable Python programs in a conda environment from using the user site packages directory. (#13635)
```

Comment:
I thought we were trying to stay away from adding language-specific options in `conda` itself now, and instead resorting to plugins. That's what prompted the inclusion of `EXTERNALLY-MANAGED` in `conda-pip`, at least, and I wonder if that project would be a better home for this type of contribution.

If that's not a good idea, then I'd lean towards having a setting that is a bit more abstract, like `environment-isolation`, that accepts a series of "features" to enable, one of them being `pyvenv.cfg`.

Reply:
That's the conflict of concept that I fear as well. Venv-style environment management is part of the conda core experience and has an outsized impact on how pip (etc) install into it.

I outlined a possible change of policy in my comment here: https://github.com/conda/conda/issues/13337#issuecomment-1972751196

Could we tweak this solution here, that when conda-pip becomes ready to roll out to users (as a default dependency for the installers), it can become the package responsible for isolating conda environments from Python site-packages? Or would that overstep the role of plugins given that environments are a core aspect of conda?

Reply:
Is the end goal to allow that kind of configuration once the behavior has changed, or will it eventually be frozen? Asking in case we want to start with a pseudo-private name like `_include_system_*` or similar.

Reply:
> [conda-pip] can become the package responsible for isolating conda environments from Python site-packages

Does that imply that an environment has to have pip (or conda-pip) in order to be isolated? Pip is not a requirement for isolation of venv environments, and consistency is nice.

Reply:
@jaimergp, I would be happy to re-write this as an `environment_isolation` setting that would accept a list of strings, `pyvenv` being one of them.

@itcarroll, would this suffice?

@jezdez, I don't think it makes sense to make it only part of the `conda-pip` package based on what @itcarroll has said.

Reply:
Changing to `environment_isolation` seems okay, even though I can't think of another package (other than python) that could benefit. I could see that someone might want the equivalent of a $PATH that only looks in the environment, as another form of isolation. So perhaps `environment_isolation` would be a conda feature and not just a python feature.